### PR TITLE
Added JsonSerializable interface and jsonSerialize() method to DataBag trait.

### PR DESCRIPTION
**Pull request by tiboitel**
- Added jsonSerializable interface to queryable objects.
- Added jsonSerialize method to DataBag trait.
- Up legendary elf from 18 to 21 on CardTest.
- Added test for jsonSerialize on CardTest.
- Added .swp on .gitignore for vim users.

### CODE REVIEW

**Comment by tiboitel** [comment]
I saw the Travis log, all tests need i update and i only updated CardTest and don't launch the others. Maybe i can do it on my free time later. As it the jsonSerialize feature works fine.

File: .php_cs [hunk 1]
```diff
@@ -1,8 +1,28 @@
 <?php
 
-$finder = Symfony\CS\Finder\DefaultFinder::create()
-    ->exclude('vendor')
-    ->in(__DIR__);
+if (class_exists('PhpCsFixer\Finder')) {    // PHP-CS-Fixer 2.x
```
Comment (guiwoda) [inline .php_cs:3]: Is this change necessary? I don't see the value in changing this.

File: composer.json [hunk 2]
```diff
@@ -9,11 +9,12 @@
...
         }
     ],
-    "require": {},
+    "require": {
+    },
     "require-dev": {
         "phpunit/phpunit": "^5.0",
         "php-vcr/php-vcr": "^1.2",
-        "friendsofphp/php-cs-fixer": "^1.12"
+        "friendsofphp/php-cs-fixer": "^2.13"
```
Comment (guiwoda) [inline composer.json:17]: Same applies here.

**Review by guiwoda — COMMENTED** [review]
The change looks good, I don't see why you'd go through all the trouble of contributing to this library instead of wrapping those objects with a serializable proxy, but I wouldn't mind merging it anyway.

Please clarify why you'd change the CS Fixer stuff or remove it if it's unnecessary.

File: composer.json (hunk 2 above)
Comment (tiboitel) [reply composer.json:17]: Can not launch PHP-cs-fix on CentOS 7 with PHP 7.2. So I updated on the latest version. I still had a problem finding the class Finder, so I installed a fix on the Friendsofphp / php-cs-fix repository. I can revert the changes if necessary.

**Comment by tiboitel** [comment]
I don't have any response since 15 days. So, do you want me to change my commit to put PHP-CS-Fixer back to version 1.12 and keep only the JsonSerializable? Thanks for your time.

**Comment by guiwoda** [comment]
I'm sorry about that, I don't really use this anymore so I'm not very invested in maintaining it.

**Comment by JustSteveKing** [comment]
Stale PR closing for now
